plugins/auth/index.js

```diff
@@ -114,14 +114,29 @@ exports.register = (server, options, next) => {
                 allowCookieToken: false,
                 allowQueryToken: true,
                 validateFunc: (token, cb) => {
-                    if (token !== pluginOptions.temporaryAccessKey) {
-                        return cb(null, false, {});
+                    if (token === pluginOptions.temporaryAccessKey) {
+                        return cb(null, true, {
+                            username: pluginOptions.temporaryAccessUser,
+                            scope: ['user']
+                        });
                     }
 
-                    return cb(null, true, {
-                        username: pluginOptions.temporaryAccessUser,
-                        scope: ['user']
-                    });
+                    // Token is an API token
+                    // using function syntax makes 'this' the request
+                    // TODO: Should log that we're authenticating a user with a token
+                    const factory = server.app.userFactory;
```

Comment:
I remember when you try to access server.app not from the request, it is kinda weird. I tried it locally, it is actually `server._parent.app.userFactory`. Can you start the server locally and confirm?